core/types/receipt.go

```diff
@@ -86,10 +86,8 @@ type receiptRLP struct {
 type receiptStorageRLP struct {
 	PostStateOrStatus []byte
 	CumulativeGasUsed uint64
-	TxHash            common.Hash
 	ContractAddress   common.Address
 	Logs              []*LogForStorage
-	GasUsed           uint64
```

Comment:
We can't blindly remove these fields, since the testnet and people on master are probably already running these. Imho we need to turn this into a second legacy receipt storage and use create yet another new one. Decoding wise we should try to decode into the new one first, then the legacy-legacy, and then the middle Gary's version.

Reply:
We also need to bump the database version to avoid people downgrading from newer versions.

Reply:
Supporting database version 4 was an open question for me. Although it never made it to an actual release, I'm happy to add support for existing databases still. Will do.